[CI] InternalAdjacencyMatrixTests testReduceRandom fails

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+multijob-unix-compatibility/os=debian/821/console

This fails reproducibly on master and 6.x (haven't tried earlier branches):
```
./gradlew :server:test \
  -Dtests.seed=3DCD311BB2EC3EFC \
  -Dtests.class=org.elasticsearch.search.aggregations.bucket.adjacency.InternalAdjacencyMatrixTests \
  -Dtests.method="testReduceRandom" \
  -Dtests.security.manager=true \
  -Dtests.locale=en-AU \
  -Dtests.timezone=America/Argentina/Tucuman
```

Faiure:
```
java.lang.AssertionError: expected:<{NWBdU=769, NWBdU&cbrWD=98, NWBdU&iOwpy=893, NWBdU&xqQsC=530, cbrWD=733, cbrWD&iOwpy=290, cbrWD&xqQsC=285, iOwpy=760, iOwpy&xqQsC=0, xqQsC=311}> but was:<{NWBdU=769, NWBdU&cbrWD=98, NWBdU&iOwpy=893, NWBdU&xqQsC=530, cbrWD=733, cbrWD&iOwpy=290, cbrWD&xqQsC=285, iOwpy=760, xqQsC=311}>
	at __randomizedtesting.SeedInfo.seed([3DCD311BB2EC3EFC:FDF09D0A94B1A7AA]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.elasticsearch.search.aggregations.bucket.adjacency.InternalAdjacencyMatrixTests.assertReduced(InternalAdjacencyMatrixTests.java:94)
	at org.elasticsearch.search.aggregations.bucket.adjacency.InternalAdjacencyMatrixTests.assertReduced(InternalAdjacencyMatrixTests.java:34)
	at org.elasticsearch.test.InternalAggregationTestCase.testReduceRandom(InternalAggregationTestCase.java:262)
```

--- a/server/src/test/java/org/elasticsearch/search/aggregations/bucket/adjacency/InternalAdjacencyMatrixTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/bucket/adjacency/InternalAdjacencyMatrixTests.java
@@ -82,8 +82,10 @@ public class InternalAdjacencyMatrixTests extends InternalMultiBucketAggregation
         final Map<String, Long> expectedCounts = new TreeMap<>();
         for (InternalAdjacencyMatrix input : inputs) {
             for (InternalAdjacencyMatrix.InternalBucket bucket : input.getBuckets()) {
-                expectedCounts.compute(bucket.getKeyAsString(),
+                if (bucket.getDocCount() > 0) {
+                    expectedCounts.compute(bucket.getKeyAsString(),
                         (key, oldValue) -> (oldValue == null ? 0 : oldValue) + bucket.getDocCount());
+                }
             }
         }
         final Map<String, Long> actualCounts = new TreeMap<>();

